tests/linearizability/model.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linearizability
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/anishathalye/porcupine"
+)
+
+type Operation int8
+
+const Get Operation = 0
+const Put Operation = 1
+
+type etcdRequest struct {
+	op      Operation
+	key     string
+	putData string
+}
+
+type etcdResponse struct {
+	getData string
+	err     error
+}
+
+type EtcdState struct {
+	Key          string
+	Value        string
+	FailedWrites map[string]bool
```

Comment:
It seems that you do not use the value at all.

```suggestion
	FailedWrites map[string]struct{}
```